packages/lib-user/src/components/shared/MainContent/MainContent.js

```diff
@@ -67,139 +46,188 @@ function MainContent({
   source = DEFAULT_SOURCE
 }) {
   const [activeTab, setActiveTab] = useState(0)
+  const [showCalendar, setShowCalendar] = useState(false)
+  const [customDateRange, setCustomDateRange] = useState([selectedDateRange.startDate, selectedDateRange.endDate])
+  
   const handleActiveTab = useCallback((tabIndex) => {
     setActiveTab(tabIndex)
   }, [])
 
+  useEffect(function updateCustomDateRange() {
+    setCustomDateRange([selectedDateRange.startDate, selectedDateRange.endDate])
+  }, [selectedDateRange])
+  
   const size = useContext(ResponsiveContext)
 
   const hoursSpent = convertStatsSecondsToHours(stats?.time_spent)
 
-  // create project options
-  let projectOptions = [
-    { label: 'ALL PROJECTS', value: 'AllProjects' },
-    ...(projects || []).map(project => ({
-      label: project.display_name,
-      value: project.id
-    }))
-  ]
-  const selectedProjectOption = projectOptions.find(option => option.value === selectedProject)
-
-  // create date range options
-  const dateRangeOptions = getDateRangeSelectOptions(source?.created_at?.substring(0, 10))
+  const { dateRangeOptions, selectedDateRangeOption } = getDateRangeSelectOptions({ created_at: source?.created_at?.substring(0, 10), selectedDateRange })
+  
+  const { projectOptions, selectedProjectOption } = getProjectSelectOptions({ projects, selectedProject })
+  
   const todayUTC = new Date().toISOString().substring(0, 10)
-  let selectedDateRangeOption = dateRangeOptions.find(option =>
-    (selectedDateRange.endDate === todayUTC) &&
-    (option.value === selectedDateRange.startDate)
-  )
+  const leastRecentDate = selectedDateRange.startDate < source?.created_at?.substring(0, 10) ? selectedDateRange.startDate : source?.created_at?.substring(0, 10)
```

Comment:
This is for a situation where the stats source has a created_at date that is more recent than the selected startDate from the preset options. 

i.e. if a group has a created_at date of April 1, 2024, and a user has "Last 7 Days" selected. If the user then goes to the custom date calendar the farthest date the calendar will go to will be April 1, 2024 (is that ok? maybe it shouldn't be?). However, if a user selects "Last 12 months" then goes to the custom date calendar it will default to "Last 12 Months" which means the farthest date the calendar will go to will be August 1, 2023.

Maybe this is unnecessarily complicated and the farthest date should just be 2015-(something-somthing when panoptes stats started).

Reply:
I like the current behavior where the chart will just be empty for those dates. However, my vote is to leave the functionality like above for now and we can revisit.